### PR TITLE
Establish default for swagger-ui.url

### LINKED ISSUE #5281: `springdoc.swagger-ui.config-url` displays invalid JSON
**Description:**
which prevents it from loading SCDF's openapi contract, when swagger ui's default url is disabled. Workaround is to manually provide `springdoc.api-docs.path` to swagger UI

**Release versions:**
```json
{
  "versions": {
    "implementation": {
      "name": "spring-cloud-dataflow-server",
      "version": "2.10.2"
    },
    "core": {
      "name": "Spring Cloud Data Flow Core",
      "version": "2.10.2"
    },
    "dashboard": {
      "name": "Spring Cloud Dataflow UI",
      "version": "3.3.2"
    },
    "shell": {
      "name": "Spring Cloud Data Flow Shell",
      "version": "2.10.2",
      "url": "https://repo.maven.apache.org/maven2/org/springframework/cloud/spring-cloud-dataflow-shell/2.10.2/spring-cloud-dataflow-shell-2.10.2.jar"
    }
  },
  "features": {
    "streams": true,
    "tasks": true,
    "schedules": true,
    "monitoringDashboardType": "NONE"
  },
  "runtimeEnvironment": {
    "appDeployer": {
      "deployerImplementationVersion": "2.9.1",
      "deployerName": "Spring Cloud Skipper Server",
      "deployerSpiVersion": "2.9.2",
      "javaVersion": "1.8.0_362",
      "platformApiVersion": "",
      "platformClientVersion": "",
      "platformHostVersion": "",
      "platformSpecificInfo": {
        "default": "kubernetes"
      },
      "platformType": "Skipper Managed",
      "springBootVersion": "2.7.9",
      "springVersion": "5.3.25"
    },
    "taskLaunchers": [
      {
        "deployerImplementationVersion": "2.8.2",
        "deployerName": "KubernetesTaskLauncher",
        "deployerSpiVersion": "2.8.2",
        "javaVersion": "1.8.0_362",
        "platformApiVersion": "v1",
        "platformClientVersion": "unknown",
        "platformHostVersion": "unknown",
        "platformSpecificInfo": {
          "namespace": "scdf",
          "master-url": "https://10.7.251.143:6443"
        },
        "platformType": "Kubernetes",
        "springBootVersion": "2.7.9",
        "springVersion": "5.3.25"
      },
      {
        "deployerImplementationVersion": "2.8.2",
        "deployerName": "KubernetesTaskLauncher",
        "deployerSpiVersion": "2.8.2",
        "javaVersion": "1.8.0_362",
        "platformApiVersion": "v1",
        "platformClientVersion": "unknown",
        "platformHostVersion": "unknown",
        "platformSpecificInfo": {
          "namespace": "scdf",
          "master-url": "https://10.96.0.1:443/"
        },
        "platformType": "Kubernetes",
        "springBootVersion": "2.7.9",
        "springVersion": "5.3.25"
      }
    ]
  },
  "monitoringDashboardInfo": {
    "url": "",
    "source": "default-scdf-source",
    "refreshInterval": 15
  },
  "security": {
    "isAuthentication": false,
    "isAuthenticated": false,
    "username": null,
    "roles": []
  }
}
```

**Custom apps:**
N/A

**Steps to reproduce:**
given the following config:
```json
springdoc:
  api-docs:
    enabled: true
  cache:
    disabled: true
  swagger-ui:
    disable-swagger-default-url: true
```
then SCDF's `springdoc.swagger-ui.config-url` displays invalid JSON, preventing swagger-ui to load `springdoc.api-docs.path`.

**Screenshots:**
N/A

**Additional context:**
N/A

### PR DESCRIPTION
**Pull request by cppwfs**
Also resolve issue if openai json is unescaped.  Test added to validate this case.

resolves #5281

### CODE REVIEW

**Review by onobc — APPROVED** [review]
LGTM - thank you very much @cppwfs - nice work!

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/support/SpringDocJsonDecodeFilterTest.java [hunk 1]
```diff
@@ -33,27 +34,49 @@
...
                 super.doFilter(request, response);
             }
         };
-        new SpringDocJsonDecodeFilter().doFilter(mockHttpServletRequest, mockHttpServletResponse, mockFilterChain);
-        assertThat(mockHttpServletResponse.getContentAsString()).isEqualTo(OPENAPI_JSON_UNESCAPED_CONTENT);
+        new SpringDocJsonDecodeFilter().doFilter(this.mockHttpServletRequest, this.mockHttpServletResponse, mockFilterChain);
+        assertThat(this.mockHttpServletResponse.getContentAsString()).isEqualTo(OPENAPI_JSON_UNESCAPED_CONTENT);
     }
+	@Test
+	public void doFilterTestUnEscaped() throws ServletException, IOException {
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/support/SpringDocJsonDecodeFilterTest.java:70]: Nice addition! Thank you

**Comment by klopfdreh** [comment]
Thanks a lot and my excuse that I caused the issue with this greedy unescape logic.

**Comment by cppwfs** [comment]
@klopfdreh Thank you for all your contributions to the project.   You just ROCK!

**Pull request merged**